### PR TITLE
fix: xai temperature len in kernel

### PR DESCRIPTION
**Pull request by JamesBrianD**
## Motivation

Fix SGL-57 https://github.com/sgl-project/sglang-jax/issues/421

## Accuracy Tests

```
sky exec sgl-jax-tpu-v6e-32-merge-two-fix-24743 -d -- “cd sglang-jax && source .venv/bin/activate  && JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache uv run python -u -m sgl_jax.launch_server --model-path /models/xai-grok-2 --trust-remote-code --tp-size=32 --device=tpu --random-seed=3 --mem-fraction-static=0.8 --chunked-prefill-size=8192 --download-dir=/dev/shm --dtype=bfloat16 --max-running-requests=128 --skip-server-warmup --page-size=128 --tokenizer-path /models/xai-grok-2/tokenizer.tok.json  --dist-init-addr=10.128.0.94:10015 --nnodes=8 --node-rank=\${SKYPILOT_NODE_RANK}”
```

```
evalscope eval  --model /models/xai-grok-2 --api-url http://127.0.0.1:30000/v1/chat/completions  --api-key EMPTY  --eval-type openai_api --datasets gpqa_diamond --eval-batch-size 198 --dataset-args ‘{“gpqa_diamond”: {“few_shot_num”: 3}}’  --generation-config ‘{“temperature”: 0.5,“top_p”:0.8,“top_k”:40, “max_tokens”: 4096}’
```
image

## Checklist

- [ ] Please use English, otherwise it will be closed.
- [ ] The purpose of the PR, or link existing issues this PR will resolve.
- [ ] The test plan, such as providing test command.
- [ ] (Optional) The necessary documentation update.

### CODE REVIEW

**Review by jimoosciuc — APPROVED** [review]
LGTM
